### DISCUSSION TITLE
What does `shouldReplace` do?

### DISCUSSION THREAD

**Discussion by Toshinaki** [pmndrs/zustand#1312] [Q&A] [answered]
I searched all  docs I could find, but got no answer.

What does the [`shouldReplace`](https://github.com/pmndrs/zustand/blob/0f4913c7ec9776f1f12e0ee16ce1b0387ab9ba76/src/middleware/immer.ts#L45) do?

What happens if set to true or false?

**Comment by dai-shi** [comment, marked as answer]
https://github.com/pmndrs/zustand/blob/0f4913c7ec9776f1f12e0ee16ce1b0387ab9ba76/src/vanilla.ts#L77-L80
If it's `true`, it will bypass the default merging behavior (`Object.assign`).

**Reply by PX-72** [reply]
How does it work when using the immer middleware?

**Reply by dbritto-dev** [reply]
@PX-72 shouldn't affect immer middleware at all because you are returning the whole store
